### PR TITLE
[AC-4708] - Impact coverage report is not diff-aware

### PR DESCRIPTION
**Pull request by yotammanor**
**Changes Introduced in [AC-4708](https://masschallenge.atlassian.net/browse/AC-4708):**

- rewrite coverage targets to print a diff-aware report.
- rewrite coverage targets to be more similar to the equivalent accelerate aliases.
- add target coverage-html-open as a separate command from coverage-html.

**To test:**
- make a
- on development, change a .py file, run `make coverage`. You'll see the entire file tree printed.
- on branch AC-4708, make that same change, run `make coverage`. You'll see just that file being printed. Also, html won't open automatically.
- change a file in tests/django_migrations/non-.py file, and it won't be printed.

### CODE REVIEW

File: web/impact/impact/models/user_role.py [hunk 1]
```diff
@@ -30,11 +30,11 @@ class UserRole(MCModel):
     STAFF = "Staff"
     TEAM = "Team"
 
-    OFFICE_HOUR_ROLES = set([AIR,
-                             PARTNER,
-                             PARTNER_ADMIN,
-                             OFFICE_HOUR_HOLDER,
-                             MENTOR])
+    OFFICE_HOUR_ROLES = {AIR,
```
Comment (mc-nathan) [inline web/impact/impact/models/user_role.py]: This change is not in Accelerate and seems irrelevant to this PR.

**Review by mc-nathan — CHANGES_REQUESTED** [review]
Remove change to user_role.py or make it clear why it's being made here and not in accelerate.

**Comment by yotammanor** [comment]
@mc-nathan reverted the utility commit, and updated testing instructions above.

**Review by mc-nathan — APPROVED** [review]

**Review by knittingarch — CHANGES_REQUESTED** [review]
Functional Issues
On the branch, the changed file doesn't show up after running `make coverage`.

Code: Looks good.
Code Quality: Clean.
Tests: Pass
Coverage: N/A

**Comment by knittingarch** [comment]
screen shot 2017-08-11 at 10 33 13 am

**Comment by mc-nathan** [comment]
I was only able to reproduce the behavior reported by @knittingarch by making a change to a file that already had full coverage in a way that it still has full coverage (e.g., add a blank line).  This seems reasonable to me.  I learned that the change that @knittingarch made was to change a True to a False in permission.py  which probably didn't change the coverage.  She's going to test a bit further.

**Review by knittingarch — APPROVED** [review]

**Pull request merged**